File: pset6/caesar.py
import cs50
import sys


if len(sys.argv) != 2:
    print('Usage: casear.py key')
    exit(1)

for c in sys.argv[1]:
    if c.isalpha():
        print('Key must be an integer')
        exit(2)

if int(sys.argv[1]) < 1:
    print('Key must be a non negative integer')
    exit(2)
else:
    k = int(sys.argv[1])

plaintext = cs50.get_string('Plaintext: ')
print('Ciphertext: ', end="")

for p in plaintext:
    if p.islower():
        print(chr((ord(p) + k - ord('a')) % 26 + ord('a')), end="")
    elif p.isupper():
        print(chr((ord(p) + k - ord('A')) % 26 + ord('A')), end="")
    else:
        print(p, end="")
print()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    int k;
    if(argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    else if (atoi(argv[1]) < 1)
    {
        printf("Key must be a non-negative integer\n");
        return 2;
    }
    else
    {
        k = atoi(argv[1]);
    }
    
    printf("Plaintext: ");
    string p = get_string(); //plaintext
    
    printf("Ciphertext: ");
    
    for (int i = 0, n = strlen(p); i < n; i++)
    {
        if(islower(p[i]))
        {
            printf("%c", (p[i] + k - 'a') % 26 + 'a');
        }
        else if(isupper(p[i]))
        {
            printf("%c", (p[i] + k - 'A') % 26 + 'A');
        }
        else
        {
            printf("%c", p[i]);
        }
    }
    
    printf("\n");
    
    return 0;
}